### DISCUSSION TITLE
Inserting images from URL, without uploading them to S3

### DISCUSSION THREAD

**Discussion by AnandChowdhary** [outline/outline#1354] [General] [answered]
Might be that I haven't figured this out yet, but is there a way to just insert and image from its URL, rather than having to upload it? I think I should be able to do it in the markdown directly somehow?

**Comment by tommoor** [comment, marked as answer]
Yes, you can do this by typing the markdown for the image – `![caption](http://....)`

**Reply by bboh** [reply]
Sorry if I am missing something but how can I insert this markdown in the outline editor (I am trying right now on getoutline.com). In contrast to typing "# " for a markdown headline that is recognized, typing "\!\[\]\(https://...\) " does insert the characters verbatim.

**Reply by tommoor** [reply]
It could be you're typing in a place where images aren't allowed – try on an empty paragraph? It definitely works.

**Reply by bboh** [reply]
Thanks. It does work. Layer 8 problem.

**Comment by soundstep** [comment]
I don't think this works anymore, typing:
```
![test](https://media.wired.com/photos/5b899992404e112d2df1e94e/master/pass/trash2-01.jpg)
```
Starting to type:
<img width="839" alt="Screenshot 2024-01-23 at 15 32 22" src="https://github.com/outline/outline/assets/133293/d40d15e4-f649-4493-bd94-426c5c8c7d27">
Adding the last parenthesis:
<img width="844" alt="Screenshot 2024-01-23 at 15 32 27" src="https://github.com/outline/outline/assets/133293/70a57031-b49c-4005-bc95-b0664dd5e15f">
That's actually a link:
<img width="941" alt="Screenshot 2024-01-23 at 15 32 44" src="https://github.com/outline/outline/assets/133293/9d4f9f4d-250a-4ea2-966f-283e81d9de6c">
Can someone confirm? Is there another way?

**Reply by tommoor** [reply]
Hi, you can paste that `![test](https://media.wired.com/photos/5b899992404e112d2df1e94e/master/pass/trash2-01.jpg)` into a doc and it will work correctly :)

**Reply by soundstep** [reply]
Oh right, that's interesting, so typing `![test](`and then pasting the URL doesn't work: it creates a link.
But writing the whole thing somewhere else (`![test](https://my-url)`), then pasting it works!
It is a little bit counter-intuitive, I would definitely add that to the doc or alter the behavior.
Thanks for that @tommoor!

**Reply by soundstep** [reply]
This is cool as I can embed stuff like [that](https://quickchart.io/sandbox#%7B%22chart%22%3A%22%7B%5Cn%20%20type%3A%20'bar'%2C%5Cn%20%20data%3A%20%7B%5Cn%20%20%20%20labels%3A%20%5B''%5D%2C%5Cn%20%20%20%20datasets%3A%20%5B%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%201'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23f00'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%202'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23f00'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%203'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23f00'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%204'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23f00'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%205'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23f00'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%206'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23ff0'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%207'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23ff0'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%208'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23ff0'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%209'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%230f0'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20label%3A%20'Part%2010'%2C%5Cn%20%20%20%20%20%20%20%20data%3A%20%5B10%5D%2C%5Cn%20%20%20%20%20%20%20%20backgroundColor%3A%20'%230f0'%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%5D%2C%5Cn%20%20%7D%2C%5Cn%20%20options%3A%20%7B%5Cn%20%20%20%20indexAxis%3A%20'y'%2C%5Cn%20%20%20%20layout%3A%20%7B%5Cn%20%20%20%20%20%20padding%3A%2040%2C%5Cn%20%20%20%20%7D%2C%5Cn%20%20%20%20scales%3A%20%7B%5Cn%20%20%20%20%20%20x%3A%20%7B%5Cn%20%20%20%20%20%20%20%20stacked%3A%20true%2C%5Cn%20%20%20%20%20%20%20%20ticks%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20display%3A%20true%2C%5Cn%20%20%20%20%20%20%20%20%20%20stepSize%3A%2010%2C%5Cn%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%20%20grid%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20drawBorder%3A%20false%2C%5Cn%20%20%20%20%20%20%20%20%20%20display%3A%20false%2C%5Cn%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20y%3A%20%7B%5Cn%20%20%20%20%20%20%20%20display%3A%20false%2C%5Cn%20%20%20%20%20%20%20%20stacked%3A%20true%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%7D%2C%5Cn%20%20%20%20plugins%3A%20%7B%5Cn%20%20%20%20%20%20legend%3A%20%7B%5Cn%20%20%20%20%20%20%20%20display%3A%20false%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20annotation%3A%20%7B%5Cn%20%20%20%20%20%20%20%20clip%3A%20false%2C%5Cn%20%20%20%20%20%20%20%20annotations%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20arrow%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%20%20type%3A%20'point'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20pointStyle%3A%20'triangle'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20backgroundColor%3A%20'%23000'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20radius%3A%207%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20xValue%3A%2075%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20yAdjust%3A%2025%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20rotation%3A%20180%2C%5Cn%20%20%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20label1%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%20%20type%3A%20'label'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20xValue%3A%2075%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20yAdjust%3A%2010%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20content%3A%20%5B'Your%20position'%5D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20font%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%20%20size%3A%2012%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%20%20weight%3A%20'bold'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20verticalLine%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%20%20type%3A%20'line'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20xMin%3A%2075%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20xMax%3A%2075%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20borderColor%3A%20'%23000'%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20borderWidth%3A%201%2C%5Cn%20%20%20%20%20%20%20%20%20%20%20%20borderDash%3A%20%5B4%2C%204%5D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%7D%2C%5Cn%20%20%7D%2C%5Cn%7D%22%2C%22width%22%3A500%2C%22height%22%3A150%2C%22version%22%3A%223%22%2C%22backgroundColor%22%3A%22%23ffffff%22%7D).

<img width="895" alt="Screenshot 2024-01-24 at 09 13 27" src="https://github.com/outline/outline/assets/133293/30db15be-0314-4e73-9c24-d456e7d344b4">
